presto-main/src/test/java/io/prestosql/operator/TestDynamicFilterSourceOperator.java

```diff
@@ -277,63 +394,105 @@ public void testCollectEmptyBuildSide()
     }
 
     @Test
-    public void testCollectTooMuchRows()
+    public void testCollectTooMuchRowsSingleColumn()
     {
-        int maxRowCount = getDynamicFilteringMaxPerDriverRowCount(pipelineContext.getSession());
-        Page largePage = createSequencePage(ImmutableList.of(BIGINT), maxRowCount + 1);
+        int maxPositionsCount = 100;
+        Page largePage = new Page(createLongSequenceBlock(0, maxPositionsCount + 1));
 
-        OperatorFactory operatorFactory = createOperatorFactory(channel(0, BIGINT));
-        verifyPassthrough(createOperator(operatorFactory),
-                ImmutableList.of(BIGINT),
-                largePage);
-        operatorFactory.noMoreOperators();
-        assertEquals(partitions.build(), ImmutableList.of(TupleDomain.all()));
+        TupleDomain<DynamicFilterId> expectedTupleDomain = TupleDomain.withColumnDomains(ImmutableMap.of(
+                new DynamicFilterId("0"), Domain.create(ValueSet.ofRanges(
+                        range(BIGINT, 0L, true, (long) maxPositionsCount, true)), false)));
+        assertDynamicFilters(maxPositionsCount, ImmutableList.of(expectedTupleDomain), ImmutableList.of(BIGINT), largePage);
     }
 
     @Test
-    public void testCollectTooMuchBytesSingleColumn()
+    public void testCollectTooMuchRowsMultipleColumns()
     {
-        long maxByteSize = getDynamicFilteringMaxPerDriverSize(pipelineContext.getSession()).toBytes();
-        Page largePage = new Page(createStringsBlock("A".repeat((int) maxByteSize + 1)));
+        int maxPositionsCount = 300;
+        Page largePage = new Page(
+                createLongSequenceBlock(0, 101),
+                createColorSequenceBlock(100, 201),
+                createLongSequenceBlock(200, 301));
 
-        OperatorFactory operatorFactory = createOperatorFactory(channel(0, VARCHAR));
-        verifyPassthrough(createOperator(operatorFactory),
-                ImmutableList.of(VARCHAR),
-                largePage);
-        operatorFactory.noMoreOperators();
-        assertEquals(partitions.build(), ImmutableList.of(TupleDomain.all()));
+        List<TupleDomain<DynamicFilterId>> expectedTupleDomains = ImmutableList.of(
+                TupleDomain.withColumnDomains(ImmutableMap.of(
+                        new DynamicFilterId("0"), Domain.create(ValueSet.ofRanges(
+                                range(BIGINT, 0L, true, 100L, true)), false),
+                        new DynamicFilterId("2"), Domain.create(ValueSet.ofRanges(
+                                range(BIGINT, 200L, true, 300L, true)), false))));
+        assertDynamicFilters(maxPositionsCount, expectedTupleDomains, ImmutableList.of(BIGINT, COLOR, BIGINT), largePage);
+    }
+
+    @Test
+    public void testCollectTooMuchBytesSingleColumn()
```

Comment:
rename to `testSingleColumnCollectMinMaxRangeWhenTooMuchBytes`

Reply:
done